File: Python/date.py
class Date:
    def iniDate(self):
        self.day = 0
        self.month = 0
        self.year = 0

    def leapYear(self):
        if self.year % 4 == 0 and self.year % 100 != 0 or self.year % 400 == 0:
            return True


    def setDay(self, d):
        if self.month == 1 or self.month == 3 or self.month == 5 or self.month == 7 or self.month == 8 or self.month == 10 or self.month == 12:
            if d < 1 or d > 31:
                return False
        if self.month == 2:
            if self.leapYear():
                if d < 1 or d > 29:
                    return False
            else:
                if d < 1 or d > 28:
                    return False
        if self.month == 4 or self.month == 6 or self.month == 9 or self.month == 11:
            if d < 1 or d > 30:
                return False
        self.day = d
        return True

    def setMonth(self, m):
        if m >= 1 and m <= 12:
            self.month = m
            return True
        else: return False


    def setYear(self,y):
        if y >= 1900 and y <= 2100:
            self.year = y
            return True
        else: return False

    def getDay(self):
        return self.day

    def getMonth(self):
        return self.month

    def getYear(self):
        return self.year

    def nextDay(self):
        aux = self.day + 1

        if not self.setDay(aux) and not self.setMonth(aux) and not self.setYear(aux):
            self.day = 1
            self.month = self.month + 1
            if not self.setDay(aux) and not self.setMonth(aux) and not self.setYear(aux):
                self.month = 1
                self.year = self.year + 1;

#main

date = Date()

date.iniDate()

print("La fecha inicial es: " + str(date.getDay()) + "/" + str(date.getMonth()) + "/" + str(date.getYear()))

date.setDay(3)
date.setMonth(11)
date.setYear(2020)

print("La fecha actual es: " + str(date.getDay()) + "/" + str(date.getMonth()) + "/" + str(date.getYear()))

date.nextDay()
print("La fecha del dia siguiente es: " + str(date.getDay()) + "/" + str(date.getMonth()) + "/" + str(date.getYear()))

File: Java/date.java
class Date{
  private int day,month,year;

  void iniDate(){
    day = 0;
    month = 0;
    year = 0;
  }

  public boolean setDay(int d){
    day = d;
    return true;
  }
  public boolean setMonth(int m){
    month = m;
    return true;
  }
  public boolean setYear(int y){
    year = y;
    return true;
  }

  public int getDay(){
    return day;
  }
  public int getMonth(){
    return month;
  }
  public int getYear(){
    return year;
  }

  private boolean leapYear(){
    return (year % 4 == 0 && year % 100 != 0 || year % 400 == 0);
  }

  public boolean rightYear(){
    return (year >= 1900 && year <= 2100);
  }
  public boolean rightMonth(){
    return (month > 0 && year <= 12);
  }
  public boolean rightDay(){
    boolean rDay;

    switch(month){
      case 2:
        if(leapYear()){
            return (day >= 1 && day <= 29);
        }else{
          return (day >= 1 && day <= 28);
        }
      case 4:
      case 6:
      case 9:
      case 11:
        return (day >= 1 && day <= 30);
      default:
        return (day >= 1 && day <= 31);
    }
  }

  public void yearDays(){
    int c = 1;

    while(c < month){
      switch(c){
        case 2: day = day + 28; break;
        case 4:
        case 6:
        case 9:
        case 11: day = day + 30; break;
        default: day = day + 31;
      }
      c++;
    }

    if(leapYear() && month > 2 ){
      day = day + 1;
    }
  }

  public void nexDay(){
    day++;

    if(!rightDay() && !rightMonth() && !rightYear()){
      day = 1;
      month ++;
      if(!rightDay() && !rightMonth() && !rightYear()){
        month = 1;
        year ++;
      }
    }
  }

  public static void main(String argv[]){
    Date d = new Date();

    d.iniDate();

    System.out.println(d.getDay() + "/" + d.getMonth() + "/" + d.getYear());

    d.setDay(3);
    d.setMonth(11);
    d.setYear(2022);

    System.out.println(d.getDay() + "/" + d.getMonth() + "/" + d.getYear());

    if(d.leapYear()){
      System.out.println("Es bisiesto");
    }

    d.yearDays();
    System.out.println("Han transcurrido: " + d.getDay() + " días en el año");

    d.nexDay();
    System.out.println("El día siguiente es: " + d.getDay());
  }
}
